core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -129,30 +143,32 @@ class RequestSendThread(val controllerId: Int,
           // if a broker goes down for a long time, then at some point the controller's zookeeper listener will trigger a
           // removeBroker which will invoke shutdown() on this thread. At that point, we will stop retrying.
           try {
-            channel.send(request)
-            receive = channel.receive()
+            selector.send(new NetworkSend(connectionId, RequestOrResponseSend.serialize(request)))
+            def findResponse = selector.completedReceives.asScala.find(_.source() == connectionId)
+            receive = SelectorUtils.pollUntilFound(selector, config.controllerSocketTimeoutMs)(findResponse)(time).getOrElse {
```

Comment:
`BlockingChannel` is still used by the old consumer, so I am not sure we can remove anything just yet. I'll double-check though.